apps/homescreen/js/app.js

```diff
@@ -321,6 +324,21 @@ const SETTINGS_VERSION = 0;
   }
 
   App.prototype = {
+    get iconSize() {
+      // If this._iconSize is undefined or 0, let's refresh the value.
+      if (!this._iconSize) {
+        for (var i = 0; i < this.icons.childNodes.length; i++) {
```

Comment:
I thought so too, but I'm told it's not a given and I've been asked to make this change before so I'm just propagating that... I'll try harder to find out for certain at some point, because I think it looks really ugly...

The main bit to change was the this.icons.children access though, that property is actually a function that builds the array, so I doubt that'd get cached.